### PR TITLE
CDAP-5664 and CDAP-5728 : DatasetId correctness verification and removal of hydrator-ext prefix

### PR DESCRIPTION
**Pull request by gokulavasan**
JIRAs : 
https://issues.cask.co/browse/CDAP-5664 (resulting from the discussion on this PR : https://github.com/caskdata/hydrator-plugins/pull/235)
https://issues.cask.co/browse/CDAP-5728

Build : http://builds.cask.co/browse/CDAP-DUT4022

### CODE REVIEW

**Comment by poornachandra** [comment]
@gokulavasan This PR adds validation for dataset names only, right? Can you file a JIRA for validating other entity names too?

**Comment by poornachandra** [comment]
LGTM

File: cdap-proto/src/test/java/co/cask/cdap/proto/id/EntityIdTest.java [hunk 1]
```diff
@@ -188,6 +188,18 @@ private void doTestToFromOldId(EntityId id) {
   }
 
   @Test
+  public void testDatasetName() {
+    // TODO: CDAP-5730: Even though dots are allowed, if we use it here, id creation will fail.
```
Comment (bdmogal) [inline cdap-proto/src/test/java/co/cask/cdap/proto/id/EntityIdTest.java]: sigh . when we fix this we'll again need upgrade. :(

I think a good thing to do would be to collect all these shortcomings of ``EntityId`` classes and not start using the new ones until we fix all these.

**Comment by gokulavasan** [comment]
@poornachandra Filed JIRA https://issues.cask.co/browse/CDAP-5732. Thanks for the review! Will merge after build passes.

File: cdap-proto/src/test/java/co/cask/cdap/proto/id/EntityIdTest.java (hunk 1 above)
Comment (gokulavasan) [inline cdap-proto/src/test/java/co/cask/cdap/proto/id/EntityIdTest.java]: @bdmogal You mean 'correct' all the shortcomings? Ya, depends on where all this class is being used. We have to do some not-so-simple parsing if we still want to support '.'s.

Comment (bdmogal) [inline cdap-proto/src/test/java/co/cask/cdap/proto/id/EntityIdTest.java]: yeah, I meant that we shouldn't start using these EntityId classes everywhere (especially places where we serialize them) before we fix these shortcomings

**Comment by gokulavasan** [comment]
Build passed. Merging.

**Pull request merged**